### PR TITLE
PW-1217: Add threeds2 page, update components to 2.5.0 version, inclu…

### PR DESCRIPTION
**Pull request by Aleffio**
…ding css frontend/adminhtml, added logic to ThreeDS2ProcessController to cancel the order in case it is not authorised

**Description**

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: app/code/community/Adyen/Payment/Helper/Data.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ class Adyen_Payment_Helper_Data extends Mage_Payment_Helper_Data
     const AFTERPAY = "afterpay";
     const CHECKOUT_CONTEXT_URL_LIVE = 'https://checkoutshopper-live.adyen.com/checkoutshopper/';
     const CHECKOUT_CONTEXT_URL_TEST = 'https://checkoutshopper-test.adyen.com/checkoutshopper/';
-    const CHECKOUT_COMPONENT_JS = 'sdk/2.1.0/adyen.js';
+    const CHECKOUT_COMPONENT_JS = 'sdk/2.4.2/adyen.js';
```
Comment (cyattilakiss) [inline app/code/community/Adyen/Payment/Helper/Data.php]: Can we go for the v 2.5.0?

File: app/code/community/Adyen/Payment/Model/Adyen/Abstract.php [hunk 2]
```diff
@@ -683,9 +683,24 @@ protected function processJsonResponse(
...
             case 'IdentifyShopper':
                 //todo implement result code handling
+                if (!empty($response['resultCode']) && !empty($response['authentication']['threeds2.fingerprintToken']) && !empty($response['paymentData'])){
+                    $payment->setAdditionalInformation('threeDS2Type', $response['resultCode']);
+                    $payment->setAdditionalInformation('threeDS2Token',
+                        $response['authentication']['threeds2.fingerprintToken']);
+                    $payment->setAdditionalInformation('threeDS2PaymentData', $response['paymentData']);
+                }
+                Mage::getSingleton('customer/session')->setRedirectUrl("adyen/process/validate3ds2");
+//                $this->_addStatusHistory($payment, $responseCode, $pspReference, $this->_getConfigData('order_status'));
```
Comment (cyattilakiss) [inline app/code/community/Adyen/Payment/Model/Adyen/Abstract.php]: remove comment

File: app/code/community/Adyen/Payment/Helper/Data.php (hunk 1 above)
Comment (Aleffio) [reply app/code/community/Adyen/Payment/Helper/Data.php]: done

File: app/code/community/Adyen/Payment/Model/Adyen/Abstract.php (hunk 2 above)
Comment (Aleffio) [reply app/code/community/Adyen/Payment/Model/Adyen/Abstract.php]: done

File: app/design/frontend/base/default/template/adyen/threeds2.phtml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            var url = '<?php echo $this->getUrl('adyen/ThreeDS2Process', array('_secure' => true)); ?>';
+
+            var promise = new Promise(function (resolve, reject) {
+                var ajaxReq = new $.ajax(url, {
+                    parameters: {isAjax: 1, method: 'POST'},
+                    success: function (result) {
+                        resolve(result);
+                    },
+                    failure: function (result) {
+                        reject(Error("There was an error with 3ds2"));
```
Comment (rkewlani) [inline app/design/frontend/base/default/template/adyen/threeds2.phtml]: can we get some useful information from result object and add some nice error message here?

File: app/design/frontend/base/default/template/adyen/threeds2.phtml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+            promise.then(function res(result) {
+                var response = JSON.parse(result);
+                if (response.authentication) {
+                    renderThreeDS2Component(response["resultCode"], response["authentication"]["threeds2.challengeToken"]);
+                } else if (response === "Authorised") {
+                    window.location = "/checkout/onepage/success";
+                } else if (response === "Refused" || response === "Error") {
+                    window.location = "/checkout/cart";
+                }
```
Comment (rkewlani) [inline app/design/frontend/base/default/template/adyen/threeds2.phtml]: will redirection to /checkout/cart will restore cart ?

Comment (Aleffio) [reply app/design/frontend/base/default/template/adyen/threeds2.phtml]: The cart restore is already done in the backend, where we call cancelThreeds2().
Here we are checking what was the result sent from the backend, and redirect to the cart page

File: app/design/frontend/base/default/template/adyen/threeds2.phtml (hunk 3 above)
Comment (Aleffio) [reply app/design/frontend/base/default/template/adyen/threeds2.phtml]: we will add proper error handling in PW-1288

**Pull request merged**